### PR TITLE
Update gradle plugins version and use better configuration patterns

### PR DESCRIPTION
**Pull request by labkey-susanh**
#### Rationale
The update to Gralde 8 requires an update to the gradle plugins for our artifactory publishing from TeamCity. I take this opportunity to also update some of the task and configuration usages to the preferred pattern for more efficient configuration.

#### Related Pull Requests
* https://github.com/LabKey/gradlePlugin/pull/168

#### Changes
* Gradle plugin version
* updates in root `build.gradle` and server `build.gradle` for task registration and project configuration syntax

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -314,99 +314,116 @@ if (BuildUtils.shouldPublish(project) || BuildUtils.shouldPublishDistribution(pr
...
-    project.task("showConfigs", group: "help", description: "Show all the configurations available in the project and their published artifacts")
-        .doLast({
-            project.configurations.forEach({
-                Configuration config ->
-                    println(config.name + ": (${config.getDescription()})")
+    project.tasks.register("showRepos", DefaultTask) {
+        task ->
+            task.group = "Help"
+            task.description = "Show the list of repositories currently in use."
+            task..doLast({
```
Comment (labkey-tchad) [inline build.gradle]: Is this `..` some special syntax or just a typo?

**Review by labkey-tchad — APPROVED** [review]

File: gradle/settings/all.gradle [hunk 2]
```diff
@@ -11,10 +11,20 @@ buildscript {
...
+//    dependencies {
+//        classpath files("../../../gradlePlugin/build/libs/gradlePlugin-${gradlePluginsVersion}.jar")
+//        classpath "org.apache.commons:commons-lang3:${commonsLang3Version}"
+//        classpath "org.apache.commons:commons-text:1.9"
+//        classpath "commons-io:commons-io:${commonsIoVersion}"
+//        classpath "org.apache.httpcomponents:httpclient:${httpclientVersion}"
+//        classpath "org.json:json:20210307"
+//        classpath "com.fasterxml.jackson.core:jackson-databind:2.12.2"
+//        classpath "org.ajoberstar.grgit:grgit-gradle:4.1.0"
+//    }
```
Comment (labkey-adam) [inline gradle/settings/all.gradle:26]: Document the purpose of this commented out block? I'm guessing similar to the settings.gradle block. Note that some of these versions are out-of-date. Should we use properties to keep them in sync as we upgrade?

File: gradle/settings/distributions.gradle [hunk 3]
```diff
@@ -11,10 +11,20 @@ buildscript {
...
+//    dependencies {
+//        classpath files("../../../gradlePlugin/build/libs/gradlePlugin-${gradlePluginsVersion}.jar")
+//        classpath "org.apache.commons:commons-lang3:${commonsLang3Version}"
+//        classpath "org.apache.commons:commons-text:1.9"
+//        classpath "commons-io:commons-io:${commonsIoVersion}"
+//        classpath "org.apache.httpcomponents:httpclient:${httpclientVersion}"
+//        classpath "org.json:json:20210307"
+//        classpath "com.fasterxml.jackson.core:jackson-databind:2.12.2"
+//        classpath "org.ajoberstar.grgit:grgit-gradle:4.1.0"
+//    }
```
Comment (labkey-adam) [inline gradle/settings/distributions.gradle:26]: Document and use properties instead of hard-coded versions?

**Review by labkey-adam — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (labkey-susanh) [reply build.gradle]: Typo. Thanks.

File: gradle/settings/all.gradle (hunk 2 above)
Comment (labkey-tchad) [reply gradle/settings/all.gradle:26]: These are to run with your locally built gradle plugin (an explanation is in `settings.gradle`).
FWIW, I like to publish the gradle plugin locally (`./gradlew publishToMavenLocal`) then add `mavenLocal()` to the list of maven repositories in these files. That way the correct dependencies all get pulled in.

Comment (labkey-susanh) [reply gradle/settings/all.gradle:26]: Yeah, that's probably the better approach, Trey.  In the interest of speed for getting installers working again, I'm going to just update comments and add properties and look to revising the suggested approach in a later PR.

**Pull request merged**